UM/View/GL/Texture.py

```diff
@@ -1,32 +1,52 @@
 # Copyright (c) 2015 Ultimaker B.V.
 # Uranium is released under the terms of the AGPLv3 or higher.
 
+from PyQt5.QtGui import QOpenGLTexture, QImage
+
 ##  A class describing the interface to be used for texture objects.
 #
 #   This interface should be implemented by OpenGL implementations to handle texture
 #   objects.
-class Texture:
-    def __init__(self):
-        pass
+class Texture(object):
+    def __init__(self, open_gl_binding_object):
+        super().__init__()
+
+        self._qt_texture = QOpenGLTexture(QOpenGLTexture.Target2D)
+        self._gl = open_gl_binding_object
+        self._file_name = None
+        self._image = None
 
     ##  Get the OpenGL ID of the texture.
     def getTextureId(self):
-        raise NotImplementedError("Should be implemented by subclasses")
+        return self._qt_texture.textureId()
 
     ##  Bind the texture to a certain texture unit.
     #
     #   \param texture_unit The texture unit to bind to.
-    def bind(self, texture_unit = 0):
-        raise NotImplementedError("Should be implemented by subclasses")
+    def bind(self, unit):
+        if not self._qt_texture.isCreated():
+            if self._file_name != None:
+                self._image = QImage(self._file_name).mirrored()
+            elif self._image is None: # No filename or image set.
+                self._image = QImage(1, 1, QImage.Format_ARGB32)
+                self._image.fill(0)
+            self._qt_texture.setData(self._image)
+            self._qt_texture.setMinMagFilters(QOpenGLTexture.Linear, QOpenGLTexture.Linear)
 
+        self._qt_texture.bind(unit)
     ##  Release the texture from a certain texture unit.
     #
     #   \param texture_unit The texture unit to release from.
```

Comment:
Ah I copied over the function, apparently the parameter was called differently. Fixed it.